json_object_private.h

```diff
@@ -100,6 +100,25 @@ void _json_c_set_last_err(const char *err_fmt, ...);
 
 extern const char *json_hex_chars;
 
+struct json_pointer_get_result {
```

Comment:
Let's add a json_pointer_private.h, to keep these a little better contained to just json_patch.c and json_pointer.c

Reply:
done in https://github.com/hawicz/json-c/tree/json_patch